src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ClaimsIdentity.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.IdentityModel.Logging;
+
+#nullable enable
+namespace Microsoft.IdentityModel.Tokens.Saml
+{
+    /// <summary>
+    /// A <see cref="SecurityTokenHandler"/> designed for creating and validating Saml Tokens. See: http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf
+    /// </summary>
+    public partial class SamlSecurityTokenHandler : SecurityTokenHandler
+    {
+        internal override ClaimsIdentity CreateClaimsIdentityInternal(SecurityToken securityToken, ValidationParameters validationParameters, string issuer)
+        {
+            return CreateClaimsIdentity(securityToken as SamlSecurityToken, validationParameters, issuer);
+        }
+
+        internal ClaimsIdentity CreateClaimsIdentity(SamlSecurityToken? samlToken, ValidationParameters validationParameters, string issuer)
+        {
+            if (samlToken == null)
```

Comment:
Clarifying question: why is `samlToken` nullable in the parameter list if this method throws when it is `null`?

Reply:
In this situation it is because the `CreateClaimsIdentityInternal` method receives a `SecurityToken` that needs to be cast to `SamlSecurityToken` and that returns a nullable value. In order to keep things simple, I decided to leave the parameter here as nullable, because otherwise I would need to push that check to the other method, and this one is already performing checks.

In practice, this path is reached only from a `ValidatedToken` issued by a `SamlSecurityTokenHandler`, so the `SecurityToken` object would always be a `SamlSecurityToken`, the cast would succeed, and the `samlToken` would never be null.

Reply:
Then is there a reason you don't directly cast it, but rather use `as`?
This is not a blocking question.